socorro/cron/jobs/ftpscraper.py

```diff
@@ -25,6 +26,45 @@
 from socorro.lib.datetimeutil import string_to_datetime
 
 
+class memoize(object):
+    """cache the return value of a method
+
+    This class is meant to be used as a decorator of methods. The return value
+    from a given method invocation will be cached on the instance whose method
+    was invoked. All arguments passed to a method decorated with memoize must
+    be hashable.
+
+    If a memoized method is invoked directly on its class the result will not
+    be cached. Instead the method will be invoked like a static method:
+    class Obj(object):
+        @memoize
+        def add_to(self, arg):
+            return self + arg
+    Obj.add_to(1) # not enough arguments
+    Obj.add_to(1, 2) # returns 3, result is not cached
+    """
+    def __init__(self, func):
+        self.func = func
+
+    def __get__(self, obj, objtype=None):
+        if obj is None:
+            return self.func
+        return partial(self, obj)
+
+    def __call__(self, *args, **kw):
+        obj = args[0]
+        try:
+            cache = obj.__cache
+        except AttributeError:
+            cache = obj.__cache = {}
+        key = (self.func, args[1:], frozenset(kw.items()))
+        if key in cache:
+            res = cache[key]
+        else:
+            res = cache[key] = self.func(*args, **kw)
+        return res
```

Comment:
This is overly complex. All you need is something like this:

```
def memoize_download(fun):
    cache = {}
    def _memoize_download(url):
        if url in cache:
            return cache[url]
        ret = fun(url)
        cache[url] = ret
        return ret
```

Or something along those lines.